File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min=prices[0]
    let max=0
    for(i=0;i<prices.length;i++){
        if(min>prices[i]){
            min=prices[i]
        }
        else if(prices[i]-min>max){
            max=prices[i]-min
        }
    }
    return max
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int minValue=prices[0];
        int profit=0;
        for(int i=0;i<=prices.Length-1;i++){
            if(prices[i]<minValue){
                minValue=prices[i];
            }
            else{
                int currentProfit=prices[i]-minValue;
                if(currentProfit>profit){
                   profit=currentProfit;
                }
            }
        }
        return profit;
    }
}